Source/WebCore/style/PropertyCascade.cpp

```diff
@@ -248,11 +248,12 @@ bool PropertyCascade::addMatch(const MatchedProperties& matchedProperties, Casca
                 return false;
             }
 
-            if (m_includedProperties.contains(PropertyType::Inherited) && current.isInherited())
+            bool currentIsInherited = CSSProperty::isInheritedProperty(current.id());
```

Comment:
If this is an expensive operation we should make sure to only do it if the result is needed. I can't tell from the structure of the code if it is always used.

Reply:
I couldn't tell either. Is it common for m_includedProperties to include both PropertyType::Inherited and PropertyType::NonInherited? @anttijk do you know?

Reply:
That's the common case. 
```
static constexpr OptionSet<PropertyType> normalProperties() { return { PropertyType::NonInherited,  PropertyType::Inherited }; }
```

Reply:
I would be surprised if this has any perf significance (and if it did we should be using the cached bit in StyleBuilder.cpp too).